src/devTools.js

```diff
@@ -187,12 +217,36 @@ class DevToolsEnhancer {
     }
   };
 
+  startWrapper = () => {
+    if (this.started || (this.socket && this.socket.getState() === this.socket.CONNECTING)) return;
+
+    if (!this.socketOptions.port) {
+      // no port provided - we should throw!
+      throw new Error('no port provided');
+    }
+    if (this.socketOptions.hostname) {
+      // hostname provided - don't look for it
+      this.start();
+    } else {
+      // obtain the hostname
+      this.urlPromise
+        .then((url) => {
+          this.socketOptions.hostname = url;
+          this.socketUrl = url;
+          this.start();
+        })
+        .catch((err) => {
+          console.log('Error obtaining socket url: ' + err.toString());
+        });
+    }
+  };
```

Comment:
this function has been added in order to resolve the promise and obtain a proper URL before connecting to the socket server

Reply:
For your new code, it would be great if you could maintain our coding conventions (eg. `return` on its own line within its own curly braces, avoid huge `else` block in favor of `return`ing in the preceding `if` block, etc.)

Reply:
ok, fixed